core/blockchain.go

```diff
@@ -2269,7 +2269,8 @@ func (bc *BlockChain) reorg(oldHead *types.Header, newHead *types.Block) error {
 	} else {
 		// len(newChain) == 0 && len(oldChain) > 0
 		// rewind the canonical chain to a lower point.
-		log.Error("Impossible reorg, please file an issue", "oldnum", oldBlock.Number(), "oldhash", oldBlock.Hash(), "oldblocks", len(oldChain), "newnum", newBlock.Number(), "newhash", newBlock.Hash(), "newblocks", len(newChain))
```

Comment:
Oh, the log message says "Impossible reorg, please file an issue", but it actually happen, why?

And it might be better to adjust the log message as well?

Reply:
This is because of the need for the taiko chain consensus mechanism.
The relate code at [line](https://github.com/taikoxyz/taiko-geth/blob/fdd905ad9d7eb0bf2aa1c0235cc36f5bf1ad412d/eth/catalyst/api.go#L402).

Reply:
Thanks @mask-pp 

It's magic that Taiko allows reorg when `len(newChain) == 0 && len(oldChain) > 0`. This event is more like "truncate" than "reorg".
BTW, I'm curious, do you know why Taiko consensus mechanism allows for reorg when `len(newChain) == 0 && len(oldChain) > 0`? // It may be an out-of-topic question.

Reply:
The consensus of the taiko chain is completely dependent on geth. If the geth chain is rolled back, the taiko chain must also be forced to roll back, so this phenomenon will occur.